pages/add/add.jsx

```diff
@@ -1,12 +1,201 @@
 import React from 'react';
+import { browserHistory } from 'react-router';
+import HintMessage from '../../components/hint-message/hint-message.jsx';
+import Service from '../../js/service.js';
+import UserData from '../../js/user-data.js';
+import env from "../../config/env.generated.json";
+
+const CreatorInputField = () => {
+  return (
+    <input name="creators" type="text" placeholder="Name" width="150" className="form-control" />
+  );
+};
 
 export default React.createClass({
+  getInitialState() {
+    return {
+      numCreatorFields: 1,
+      loggedInStatusChecked: false
+    };
+  },
+  componentWillMount() {
+    console.log(`UserData.userLoggedInStatus.get() = `, UserData.userLoggedInStatus.get());
+  },
+  componentDidMount() {
+    let loggedInParam = this.props.router.location.query.loggedin;
+
+    if ( !loggedInParam || loggedInParam.toString().toLowerCase() === `true` ) {
```

Comment:
This code (and the verify and update functions) should live in their own file that you can require in, and then call as
```
const userStatus = require('./userstatus')
...
componentDidMount() {
  if (this.props.router.location.query) {
    userstatus.handle(this.props.router.location.query);
  }
}
```
And then because `Service` is shared, anything `userstatus` sets pertaining to the user status will be in effect inside this class, too.